packages/itwinui-css/src/surface/surface.scss

```diff
@@ -22,5 +22,43 @@
 
 // Public facing surface component.
 @mixin iui-surface-component {
+  --iui-surface-padding-inline: var(--iui-size-s);
+  --iui-surface-body-padding: var(--iui-surface-padding-inline);
+
   @include iui-surface;
+  display: flex;
+  flex-direction: column;
+  height: var(--iui-surface-height);
+}
+
+@mixin iui-surface-header-footer($type) {
+  display: flex;
+  flex-shrink: 0;
+  align-items: center;
+  overflow: hidden;
+
+  @if ($type == 'header') {
+    border-bottom: 1px solid var(--iui-surface-border-color);
+    padding: calc(var(--iui-surface-padding-inline) * 0.5) var(--iui-surface-padding-inline);
+  }
+
+  @if ($type == 'footer') {
+    border-top: 1px solid var(--iui-surface-border-color);
+  }
+}
+
+@mixin iui-surface-header {
+  @include iui-surface-header-footer(header);
+}
+
+@mixin iui-surface-footer {
```

Comment:
I'm wondering if this footer even needs to exist. It might make more sense to have a `<Surface.Divider>` so that users can do whatever they want within each section.

Reply:
Removed the footer & using `:not(:last-child)` to apply a bottom border on `.iui-surface-body`.

Reply:
`:last-child` won't be reliable, as the user might want to place something else at the end. I really think a `Surface.Divider` component is the most flexible way to do this.

You might even be able to remove the `Surface.Header` if you do this (not tested).
```tsx
<Surface>
  <Surface.Content padded>...header stuff...</Surface.Content>
  <Surface.Divider />
  <Surface.Content padded>...header stuff...</Surface.Content>
  <Surface.Divider />
  <Surface.Content>...footer stuff...</Surface.Content>
</Surface>
```

Reply:
I removed the bottom border and introduced a new `.iui-divider` component.  I did not get rid of `.iui-surface-header` & `.iui-surface-body` for a generic content area because I _think_ the 2 require different CSS:
- Body needs to support overflow.
- Body needs to have `flex-grow: 1`.